kroma-validator/l2_output_submitter.go

```diff
@@ -268,30 +298,91 @@ func (l *L2OutputSubmitter) CalculateWaitTime(ctx context.Context, nextBlockNumb
-func (l *L2OutputSubmitter) HasEnoughDeposit(ctx context.Context) (bool, error) {
+// CanSubmitOutput checks if the validator satisfies the condition to submit L2Output.
+func (l *L2OutputSubmitter) CanSubmitOutput(ctx context.Context, outputIndex *big.Int) (bool, error) {
 	cCtx, cCancel := context.WithTimeout(ctx, l.cfg.NetworkTimeout)
 	defer cCancel()
 	from := l.cfg.TxManager.From()
-	balance, err := l.valpoolContract.BalanceOf(optsutils.NewSimpleCallOpts(cCtx), from)
-	if err != nil {
-		return false, fmt.Errorf("failed to fetch deposit amount: %w", err)
+	if l.IsValPoolTerminated(outputIndex) {
+		validatorStatus, err := l.getValidatorStatus(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		validatorActivated, err := l.tryActivateValidator(validatorStatus)
+		if err != nil {
+			return false, err
+		}
+		if !validatorActivated {
+			return false, nil
+		}
+
+		if validatorStatus != StatusActive {
+			l.log.Warn("validator has activated, but currently has not enough tokens", "currentStatus", validatorStatus)
+			return false, nil
+		}
+
+		if isInJail, err := l.isInJail(ctx); err != nil {
+			return false, err
+		} else if isInJail {
+			l.log.Warn("validator is in jail")
+			return false, nil
+		}
+	} else {
+		balance, err := l.valpoolContract.BalanceOf(optsutils.NewSimpleCallOpts(cCtx), from)
+		if err != nil {
+			return false, fmt.Errorf("failed to fetch deposit amount: %w", err)
+		}
+		l.metr.RecordDepositAmount(balance)
+
+		if balance.Cmp(l.requiredBondAmount) == -1 {
+			l.log.Warn(
+				"deposit is less than bond attempt amount",
+				"requiredBondAmount", l.requiredBondAmount,
+				"deposit", balance,
+			)
+			return false, nil
+		}
+
+		l.log.Info("deposit amount and bond amount", "deposit", balance, "bond", l.requiredBondAmount)
 	}
-	l.metr.RecordDepositAmount(balance)
 
-	if balance.Cmp(l.requiredBondAmount) == -1 {
-		l.log.Warn(
-			"deposit is less than bond attempt amount",
-			"requiredBondAmount", l.requiredBondAmount,
-			"deposit", balance,
-		)
+	return true, nil
+}
+
+// tryActivateValidator checks if the validator can activate and tries to activate it if not activated yet.
+func (l *L2OutputSubmitter) tryActivateValidator(validatorStatus uint8) (bool, error) {
+	if validatorStatus <= StatusRegistered {
+		l.log.Warn("validator cannot activate yet", "currentStatus", validatorStatus)
 		return false, nil
 	}
-	l.log.Info("deposit amount", "deposit", balance)
-
+	if validatorStatus >= StatusInactive {
+		l.log.Info("validator has already been activated, no need to activate again", "currentStatus", validatorStatus)
+		return true, nil
+	}
+	data, err := l.valManagerAbi.Pack("activateValidator")
+	if err != nil {
+		return false, fmt.Errorf("failed to create activate validator transaction data: %w", err)
+	}
+	if txResponse := l.activateValidatorTx(data); txResponse.Err != nil {
+		return false, txResponse.Err
+	}
+	l.log.Info("validator successfully activated")
 	return true, nil
```

Comment:
Why don't we make this func simple?
Logics for checking status can be removed in this function.
Let's make this function just do activating validator without any checking status. 
And call this function in initialize L2 output submitter if the status is `READY`.
This is a function that is called on a one-time basis.